### PR TITLE
doc: add documentation for OpenID connection authentication with PiWeb server and PiWeb cloud

### PR DESCRIPTION
**Pull request by czjlorenz**

### CODE REVIEW

File: _pages/API/01-general.md [hunk 1]
```diff
@@ -270,13 +270,55 @@ Content-Type: application/json; charset=utf-8
...
+
+```json
+{"openIdAuthority":"https://issuerHost:port/basePath"}
+```
+
+The OpenID issuer URL can be used together with the [OpenID connect](https://openid.net/connect/) specifications to obtain an access token. You can use
+a OpenID connect client library to help dealing with all the details. Following a short description together with the PiWeb specifics.
+
+First use [OpenID connect discovery](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig) to get the URL of the authorization and
+token endpoints. The you can use the [OpenID connect](https://openid.net/specs/openid-connect-core-1_0.html) specification to get an access token.
```
Comment (czthiele) [inline _pages/API/01-general.md:303]: Typo: The**n** you can

File: _pages/API/01-general.md [hunk 2]
```diff
@@ -270,13 +270,55 @@ Content-Type: application/json; charset=utf-8
 if your chosen tool or script language does not work directly with present encoding.
 
 <h2 id="{{page.sections['general']['secs']['security'].anchor}}">{{page.sections['general']['secs']['security'].title}}</h2>
-Access to PiWeb Server service might require authentication. Authentication can be either *basic authentication*, based on username and password, or *Windows authentication* based on Active Directory integration.
+Access to PiWeb Server service might require authentication. Authentication can be either *basic authentication*, based on username and password or *Windows authentication* based on Active Directory integration
```
Comment (czthiele) [inline _pages/API/01-general.md:273]: Maybe replace the first 'or' with a comma to not have two 'or's in this sentence.

File: _pages/SDK/v7.1/022-basics/0225-basics-security.md [hunk 3]
```diff
@@ -49,3 +50,23 @@ DataServiceClient.AuthenticationContainer = authContainer;
 {% endhighlight %}
 This tells the ServiceClient to use Windows Authentication (ActiveDirectory). From now on the client will use your Windows user account for authentication. You do not have to specify credentials as the client will check and fetch your information from the organization directory. <br>
 Remember that your authentication mode has to match the mode set in the server settings.
+
+{{ site.headers['example'] }} Authenticate using OpenID connect
+
+{% highlight csharp %}
+// use OAuthHelper to get the OIDC access token for the service
+var oAuthTokenCredential = OAuthHelper.GetAuthenticationInformationForDatabaseUrl(serverUri.AbsoluteUri, requestCallback: OAuthRequestCallback);
```
Comment (czthiele) [inline _pages/SDK/v7.1/022-basics/0225-basics-security.md:58]: Should we encourage the use of the async variant, so GetAuthenticationInformationForDatabaseUrlAsync instead?

**Review by czthiele — APPROVED** [review]

File: _pages/API/01-general.md (hunk 1 above)
Comment (czjlorenz) [reply _pages/API/01-general.md:303]: danke. erledigt

File: _pages/API/01-general.md (hunk 2 above)
Comment (czjlorenz) [reply _pages/API/01-general.md:273]: erledigt

File: _pages/SDK/v7.1/022-basics/0225-basics-security.md (hunk 3 above)
Comment (czjlorenz) [reply _pages/SDK/v7.1/022-basics/0225-basics-security.md:58]: done

**Pull request merged**